File: ex1013.java
class Solution {
    public boolean canThreePartsEqualSum(int[] A) {
        // Get the sum of the array
        int sum = 0;
        for (int a : A)    sum += a;
        
        // Get the sum of each partition
        if (sum % 3 != 0)    return false;
        int part = sum / 3;
        
        int s = 0, cnt = 0;
        for (int i = 0; i < A.length; i++) {
            s += A[i];
            if (s == part) {
                s = 0;    cnt++;
            }
        }
        
        return (cnt == 3) ? true : false;
    }
}

File: code/ex1013.cpp
class Solution {
public:
    bool canThreePartsEqualSum(vector<int>& A) {
        int n = A.size();
        int sum = accumulate(A.begin(), A.end(), 0);
        if (sum % 3 != 0)    return 0;
        int part = sum / 3;
        
        int sumi = 0;
        for (int i = 0; i < n - 2; i++) {
            sumi += A[i];
            if (sumi == part) {
                int sumj = 0;
                for (int j = i + 1; j < n - 1; j++) {
                    sumj += A[j];
                    if (sumj == part) {
                        int sumk = 0;
                        for (int k = j + 1; k < n; k++)
                            sumk += A[k];
                        if (sumk == part)    return 1;
                    }
                }
            }
        }
        return 0;
    }
};